makefile

```diff
@@ -103,6 +104,10 @@ check_deps:
 	@freetype-config --cflags >/dev/null 2>&1 || (echo "Error: missing dependency libfreetype. Try installing libfreetype development package (e.g: libfreetype6-dev)" && false)
 	@pkg-config --cflags zlib >/dev/null 2>&1 || (echo "Error: missing dependency zlib. Try installing zlib development package (e.g: zlib-devel)" && false)
 
+# This might fail on non GNU systems as sed -i in GNU sed only
+insert_hash:
+	@sed -i 's/commit_hash = ".*"/commit_hash = "$(GIT_HASH)"/' src/argparse.cpp
```

Comment:
I think we need to add a dependency for src/argparse.cpp to ensure insert_hash is done before.
A single line:
`src/argparse.cpp: insert_hash`
should do it